samwell/sam/sambuilder.py

```diff
@@ -420,18 +438,29 @@ def to_path(self,
                 path = Path(fp.name)
 
         with NamedTemporaryFile(suffix=".bam", delete=True) as fp:
+            file_handle: IO
+            if self.sort_order is SamOrder.Unsorted:
+                file_handle = path.open('w')
+            else:
+                file_handle = fp.file
 
-            with sam.writer(fp.file,  # type: ignore
+            with sam.writer(file_handle,  # type: ignore
                             header=self._samheader,
                             file_type=sam.SamFileType.BAM) as writer:
                 for rec in self._records:
                     if pred(rec):
                         writer.write(rec)
 
-            pysam.sort("-o", str(path), fp.name)
-            if index:
-                pysam.index(str(path))
+            default_samtools_opt_list = ["-o", str(path), fp.name]
 
+            if self.sort_order is SamOrder.Unsorted:
+                file_handle.close()
```

Comment:
wouldn't we always close this ?

Reply:
I could be wrong, but I think that the `with` statement is already handling the context for the `fp.file` that we're setting to file handle when we're not unsorted. I'm slightly hesitant to close it within the with statement because the tempfile documentation says that the file is deleted as soon as it is closed. But closing it as you suggest still seems to be passing tests, so maybe I'm just missing something.